combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeUARTMessage.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Project: Combustion Inc. Android Framework
+ * File: MessageType.kt
+ * Author: https://github.com/miwright2
+ *
+ * MIT License
+ *
+ * Copyright (c) 2022. Combustion Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package inc.combustion.framework.ble.uart.meatnet
+
+/**
+ * Representation of Combustion BLE Node UART request. This top-level representation
+ * is useful for decoding multiple UART messages from a single notification that can
+ * be both NodeRequest and NodeResponse types.
+ */
+internal open class NodeUARTMessage {
+    companion object {
+        /**
+         * Returns an array of NodeUARTMessage objects (both responses and requests)
+         * from the given data from a UART notification.
+         */
+        fun fromData(data : UByteArray) : List<NodeUARTMessage> {
+            val messages = mutableListOf<NodeUARTMessage>()
+
+            var numberBytesRead = 0
+
+            while(numberBytesRead < data.size) {
+                val bytesToDecode = data.copyOfRange(numberBytesRead, data.size)
+
+                // Try to parse a NodeResponse
+                val response = NodeResponse.responseFromData(bytesToDecode)
+                if (response != null) {
```

Comment:
Minor nitpick: Rather than check for null, you can use the `let` operator, but this probably fine.
 
`b?.let {
    // If b is not null.
} ?: run {
    // If b is null.
}`

Reply:
I tried to make this change but it turns out doing so means you can't `break` from the encompassing `while()` loop. Apparently the `run {}` happens in a different lambda function context.